tests/config/test_email.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tempfile
+
+from synapse.util.stringutils import random_string
+
+from tests import unittest
+
+
+class EmailConfigTestCase(unittest.HomeserverTestCase):
+    def prepare(self, reactor, clock, hs):
+        self.hs = hs
+
+    def test_loading_missing_templates(self):
+        # Use a temporary directory that exists on the system, but that isn't likely to
+        # contain template files
+        tmp_dir = tempfile.gettempdir()
+
+        # Attempt to load an HTML template from our custom template directory
+        template = self.hs.config.read_templates(["sso_error.html"], tmp_dir,)[0]
+
+        # If no errors, we should've gotten the default template instead
```

Comment:
It would be nice to extend this test to create the a file to ensure that overriding works, you can use [`tempfile.NamedTemporaryFile`](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile) to help with this.

(I don't think any of the helpers let you provide a name for the file -- so that it can generate a unique on each time -- so you'll need to use `returned_file.name` or something when loading the template, but it should still test the custom directory stuff.)

Reply:
I've added a new test for this.